component.js

```diff
@@ -95,19 +95,21 @@ module.exports.withDefaults = factory;
 function factory (options) {
   var debug;
   options = options || {};
-  var _shouldComponentUpdate = options.shouldComponentUpdate;
+  var _componentWillMount = options.componentWillMount ||
+                            componentWillMount;
+  var _shouldComponentUpdate = options.shouldComponentUpdate ||
+                               shouldComponentUpdate.withDefaults(options);
+  var _componentWillReceiveProps = options.componentWillReceiveProps ||
```

Comment:
> I don't know if we want to have this as an overriable. Would it be necessary to deactivate this behaviour? 

I made this configurable through `withDefaults` as I thought  `withDefaults` was intended to let you fully reconfigure omniscients behavior, I can tell if anyone would want to deactivate this behavior, but I think it's unlikely.

> if not, instead of overriding the mixin, you would simply pass in a new mixin as all life cycle mixins are guaranteed to be triggered.

As far as I recall react throws if two mixins have colliding lifetime method, I could be wrong, but I can give it a try. It could be that has changed between react versions.

> Or does ordering matter in this case?

I don't really understand what you mean here. If you referring to order of lines above, it does not.